src/coreclr/jit/importercalls.cpp

```diff
@@ -3595,6 +3597,14 @@ GenTree* Compiler::impIntrinsic(CORINFO_CLASS_HANDLE    clsHnd,
                 break;
             }
 
+            case NI_System_Runtime_CompilerServices_RuntimeHelpers_WriteBarrier:
+            {
+                GenTree* val = impPopStack().val;
+                GenTree* dst = impPopStack().val;
+                retNode      = gtNewStoreIndNode(TYP_REF, dst, val, GTF_IND_TGT_HEAP);
+                break;
+            }
```

Comment:
I think I don't quite understand the flow here. If both the inlined version does `dst = src`, and if we import it as such as well, then why does the managed code not just do that directly instead of using `RuntimeHelpers.WriteBarrier`?